getnacelle/nuxt-reference-store
# fix: prevent sdk methods from being removed

### WHY are these changes introduced?

Fixes [[LS-1402](https://nacelle.atlassian.net/browse/LS-1402)] 

`useFetch` from `@nuxtjs/composition-api` is removing methods from objects returned from `setup()`. This is causing an issue since `nacelleSdk` is being injected and returned in `layouts/default.vue` and passed into the Space Provider with no `data` methods.

### WHAT is this pull request doing?

Content and `useFetch` are moved into a separate component (`SiteContainer.vue`) so that `nacelleSdk` can still be injected, returned, and passed to the Space Provider in `layouts/default.vue`.

### How to Test

1. Run `npm i`
2. Run `npm run dev`
3. Go to your localhost and navigate the site. See that products and collections are loading without the need for a refresh.

Can also run `npm run generate` and `npm run serve` to verify that static generation still works.

Additionally, preview mode will still work with these changes. (minus the bug detailed in https://nacelle.atlassian.net/browse/LS-1333)
